python/mspasspy/db/schema.py

```diff
@@ -597,5 +597,83 @@ def set_writeable(self, key):
             raise MsPASSError(key + ' is not defined', 'Invalid')
         self._main_dic[key]['readonly'] = False
 
+
+'''
+check if a mspass.yaml file is valid or not
+'''
+def is_basic_type(s):
+    if s in ["ObjectID","int","float","double","complex","bool","string","list","dict","bytes"]:
+        return True
+    return False
+
+def get_all_db_name(schema_dic):
+    db_name_list = []
+    for db in schema_dic:
+        db_name_list.append(db)
+    return db_name_list
+
+def get_all_collection_name_by_db(schema_dic, db):
+    collection_name_list = []
+    if db not in schema_dic:
+        return collection_name_list
+    for c in schema_dic[db]:
+        collection_name_list.append(c)
+    return collection_name_list
+
 def _check_format(schema_dic):
-    pass
+    db_name_list = get_all_db_name(schema_dic)
+    collection_name_list = []
+    for db_name in db_name_list:
+        collection_name_list += get_all_collection_name_by_db(schema_dic, db_name)
+    
+    type_attribute_schema = schema.Schema({
+        'type':schema.And(str, lambda s: is_basic_type(s)),
+        schema.Optional('reference'):schema.And(str, lambda s: s in collection_name_list),
+        schema.Optional('concept'):str,
+        schema.Optional('aliases'):schema.Or(list, str),
```

Comment:
I am not sure that a `str` is supported. Maybe we should automatically convert `str` into a single element `list` here.

Reply:
because I notice our mspass.yaml template has attributes that their aliases is just a string(see Database:wf_TimeSeries:schema:starttime_shift). I am not sure if it is expected here.

Reply:
hmmm... so we do support that. Good catch. I think that was a unintentional mistake, but I just checked and found that we do support str.

Reply:
I am taking that back. Turns out some methods support that but some others don't:
```
>>> from mspasspy.db.schema import DatabaseSchema
>>> dbschema = DatabaseSchema()
>>> dbschema.wf_TimeSeries.aliases('starttime_shift')
't0shift'
>>> dbschema.wf_TimeSeries.has_alias('starttime_shift')
True
>>> dbschema.wf_TimeSeries.unique_name('t0shift')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/iwang/.local/lib/python3.6/site-packages/mspasspy/db/schema.py", line 285, in unique_name
    raise MsPASSError(aliasname + ' is not defined', 'Invalid')
mspasspy.ccore.utility.MsPASSError: t0shift is not defined
>>> dbschema.wf_TimeSeries.is_alias('t0shift')
False
>>> dbschema.wf_TimeSeries.is_defined('t0shift')
False
```

Reply:
btw, this is what happened:
```
>>> dbschema.wf_TimeSeries._alias_dic
{'nsamp': 'npts', 'wfdisc.nsamp': 'npts', 'dt': 'delta', 't0': 'starttime', 'time': 'starttime', 't': 'starttime_shift', '0': 'starttime_shift', 's': 'starttime_shift', 'h': 'starttime_shift', 'i': 'starttime_shift', 'f': 'starttime_shift', 'file.dir': 'dir', 'wfdisc.dir': 'dir', 'wfprocess.dir': 'dir', 'file.dfile': 'dfile', 'wfdisc.dfile': 'dfile', 'wfprocess.dfile': 'dfile', 'file.foff': 'foff', 'wfdisc.foff': 'foff', 'wfprocess.foff': 'foff'}
```
The `t0shift` is taken apart into characters.